src/Data/List/Membership/Setoid.agda

```diff
@@ -32,14 +32,30 @@ x ∈ xs = Any (x ≈_) xs
 _∉_ : A → List A → Set _
 x ∉ xs = ¬ x ∈ xs
 
+∃[x∈xs] : ∀ {p} (P : Pred A p) → Pred (List A) _
+∃[x∈xs] P xs = ∃ λ x → x ∈ xs × P x
+
+∃∈-syntax : ∀ {p} (P : Pred A p) → Pred (List A) _
+∃∈-syntax = ∃[x∈xs]
+
+syntax ∃∈-syntax (λ x → P) xs = ∃[ x ∈ xs ] P
+
+∀[x∈xs] : ∀ {p} (P : Pred A p) → Pred (List A) _
+∀[x∈xs] P xs = ∀ {x} → x ∈ xs → P x
+
+∀∈-syntax : ∀ {p} (P : Pred A p) → Pred (List A) _
+∀∈-syntax = ∀[x∈xs]
+
+syntax ∀∈-syntax (λ x → P) xs = ∀[ x ∈ xs ] P
```

Comment:
Having looked through it again, my personal preference would be to simply remove these new definitions:
 - We've added 14 lines of code here in order to slightly shorten (not even remove) ~10 lines of code elsewhere.
 - The gains are pretty minimal `∀[x∈xs] P xs` (10 symbols) vs `∀ {x} → x ∈ xs → P x` (12 symbols)
 - I don't see the usage of these definitions increasing in the future. They're being used as a compatibility bridge to `All` and `Any`. We've crossed the bridge by proving the lemmas.
 - While the names are better now, I'm still super confused by `∃[ x ∈ xs ] P x` which looks to me like `(∃[ x ∈ xs ] P) x` but of course actually parses as `∃[ x ∈ xs ] (P x)` which is very non-intuitive to me.

However, it's not a hill I'm determined to make a stand on. If you're keen on them @jamesmckinna, then I'll bow out.

Reply:
Actually it's quite hard to get a sense of how much they are used because of all the other changes. Given that this is the only controversial part, maybe it might be worth considering splitting out the introduction of this new syntax into a new PR?

Reply:
Have removed all the new syntax, and will reconsider in a new PR in the light of all the various comments on it form the three of us.